### PR TITLE
Passthrough `CSI 3 J` in Conpty

### LINKED ISSUE #2715: RIS doesn't propagate through ConPTY to clear connected terminal
# Environment

```none
Windows build number: Win32NT 10.0.18362.0 Microsoft Windows NT 10.0.18362.0
Windows Terminal version (if applicable): 0.4.2382.0
Ubuntu WSL version: 1804.2019.521.0
```

# Steps to reproduce

Enter `printf "\033c"` while using WSL in Windows Terminal

# Expected behavior

Console should be cleared, as when using any standard Linux terminal

# Actual behavior

Nothing changes.

Could be related to #140, but it seems like that has been fixed.

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

Conpty doesn't need `CSI 3 J`, it doesn't have a scrollback. The terminal that's connected should use that. This makes conpty pass it through, like other sequences that conpty has no need for.

## PR Checklist
* [x] Closes #2715
* [x] I work here
* [x] Tests added/passed
* [n/a] Requires documentation to be updated

### CODE REVIEW

**Comment by j4james** [comment]
I don't want sound like I'm objecting to this PR, because it does at least improve the situation, but "passthrough" as a solution is almost never going to solve the problem completely. In this particular case you're fixing the `CSI 3 J` escape sequence, but any app that clears the scrollback through the console APIs is still not going to work.

In particular, issue #3126 (which was a closed as a dup of #2715) is not fixed by this PR. So if you close #2715 then you'll need to reopen #3126.

**Comment by miniksa** [comment]
> I don't want sound like I'm objecting to this PR, because it does at least improve the situation, but "passthrough" as a solution is almost never going to solve the problem completely. In this particular case you're fixing the `CSI 3 J` escape sequence, but any app that clears the scrollback through the console APIs is still not going to work.
> 
> In particular, issue #3126 (which was a closed as a dup of #2715) is not fixed by this PR. So if you close #2715 then you'll need to reopen #3126.

I reactivated 3126, @j4james. Thanks for the consideration.

File: src/terminal/adapter/adaptDispatch.cpp [hunk 1]
```diff
@@ -1505,6 +1512,17 @@ bool AdaptDispatch::HardReset()
     // delete all current tab stops and reapply
     _pConApi->PrivateSetDefaultTabStops();
 
+    // GH#2715 - If all this succeeded, but we're in a conpty, return `false` to
+    // make the state machine propogate this RIS sequence to the connected
+    // terminal application. We've reset our state, but the connected terminal
+    // might need to do more.
+    bool isPty = false;
+    _pConApi->IsConsolePty(isPty);
```
Comment (miniksa) [inline src/terminal/adapter/adaptDispatch.cpp]: This really feels like a function that should have been returning the bool...

Comment (j4james) [reply src/terminal/adapter/adaptDispatch.cpp]: Big +1. This is one of things I was hoping to clean up when moving to exceptions in place of return values for the error handling (which is still on my long term todo list). But some of these getters aren't even using the return value, so they don't actually need to have the exception handling in place - it should be an easy update.

**Pull request merged**